src/Assets/Bugsnag/Scripts/BugsnagSettingsObject.cs

```diff
@@ -79,7 +79,16 @@ public Configuration GetConfig()
 
             config.BreadcrumbLogLevel = GetLogTypeFromLogLevel( BreadcrumbLogLevel );
             config.Context = Context;
-            config.DiscardClasses = DiscardClasses;
+            foreach(string discardedClass in DiscardClasses){
+                try
+                {
+                    config.DiscardClasses.Add(new System.Text.RegularExpressions.Regex(discardedClass));
+                }
+                catch (Exception e)
+                {
+                    Debug.LogError("Regex error adding discard class: " + e.Message);
+                }
```

Comment:
Let's make this more specific:

```suggestion
                catch (ArgumentException e)
                {
                    Debug.LogError("Invalid Regex pattern for discard class: " + e.Message);
                }
```